tests/sentry/filters/test_web_crawlers.py

```diff
@@ -44,3 +44,7 @@ def test_filters_slack(self):
 
         data = self.get_mock_data('Slackbot 1.0(+https://api.slack.com/robots)')
         assert self.apply_filter(data)
+
+    def test_filters_calypso_appcrawler(self):
+        data = self.get_mock_data('Calypso AppCrawler')
```

Comment:
Is this really the entire User Agent? Seems odd.

Reply:
No, the user agent appears to be `Mozilla/5.0 (Linux; Android 6.0.1; Calypso AppCrawler Build/MMB30Y; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/53.0.2785.124 Mobile Safari/537.36 OYSTER_HYBRID_APP`. However, much of that might change, I would think, so I'd assume we only want to be searching on the stable part, ie `Calypso AppCrawler`.

If I understand correctly what we do with it, it should match a partial. We use `re.search`, and the python docs say, "re.search() checks for a match anywhere in the string."

Reply:
Correct, but the test should use the full User Agent string, since that's what we want to make sure we match.

I'm explicitly commenting this on the test. What you have in `src/sentry/filters/web_crawlers.py` in theory is ok, if it matches that User Agent string. Right now, your test isn't proving that.

Reply:
Okay, patterning after `test_does_not_filter_chrome` I included everything except the bit which is customer specific (the `OYSTER_HYBRID_APP` bit).